sdk/keyvault/Azure.Management.KeyVault/tests/VaultOperationsTestsBase.cs

```diff
@@ -0,0 +1,194 @@
+        public string vaultName { get; internal set; }
+        public VaultProperties vaultProperties { get; internal set; }
+
+
+        public VaultsClient VaultsClient { get; set; }
+        public ResourcesClient ResourcesClient { get; set; }
+        public ResourceGroupsClient ResourceGroupsClient { get; set; }
+        public UsersClient GraphUsersClient { get; set; }
+        public ProvidersClient ResourceProvidersClient { get; set; }
+
+
+        protected VaultOperationsTestsBase(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        protected async Task Initialize()
+        {
+            if (Mode == RecordedTestMode.Playback && Recording.IsTrack1SessionRecord())
+            {
+                this.tenantId = TestEnvironment.TenantIdTrack1;
+                this.subscriptionId = TestEnvironment.SubscriptionIdTrack1;
+                this.applicationId = TestEnvironment.ApplicationIdTrack1;
+            }
+            else
+            {
+                this.tenantId = TestEnvironment.TenantId;
+                this.subscriptionId = TestEnvironment.SubscriptionId;
+                this.applicationId = TestEnvironment.ClientId;
+            }
+
+            VaultsClient = GetVaultsClient();
+            ResourcesClient = GetResourcesClient();
+            ResourceGroupsClient = GetResourceGroupsClient();
+            ResourceProvidersClient = GetResourceProvidersClient();
+
+            if (Mode == RecordedTestMode.Playback)
+            {
+                this.objectId = Recording.GetVariable(ObjectIdKey, string.Empty);
+            }
+            else if (Mode == RecordedTestMode.Record)
+            {
+                var spClient = new Graph.Rbac.ServicePrincipalsClient(this.tenantId, TestEnvironment.Credential);
+                var servicePrincipalList = spClient.ListAsync($"appId eq '{this.applicationId}'");
+                await foreach (var servicePrincipal in servicePrincipalList)
+                {
+                    this.objectId = servicePrincipal.ObjectId;
+                    Recording.GetVariable(ObjectIdKey, this.objectId);
+                    break;
+                }
+            }
+            var provider = (await ResourceProvidersClient.GetAsync("Microsoft.KeyVault")).Value;
+            this.location = provider.ResourceTypes.Where(
+                (resType) =>
+                {
+                    if (resType.ResourceType == "vaults")
+                        return true;
+                    else
+                        return false;
+                }
+                ).First().Locations.FirstOrDefault();
+
+            rgName = Recording.GenerateAssetName("sdktestrg");
+            await ResourceGroupsClient.CreateOrUpdateAsync(rgName, new Resource.Models.ResourceGroup(location));
+            vaultName = Recording.GenerateAssetName("sdktestvault");
+
+            tenantIdGuid = new Guid(tenantId);
+            objectIdGuid = objectId;
```

Comment:
Why do you need the objectIdGuid ?

Reply:
Good catch, remove it since it is not used.